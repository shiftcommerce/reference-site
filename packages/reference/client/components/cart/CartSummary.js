// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Libs
import { calculateCartSummary } from '../../lib/calculateCartSummary'
import { fixedPrice } from '../../lib/fixedPrice'

// Objects
import Button from '../../objects/Button'

class CartSummary extends Component {
  shippingText () {
    if (typeof window === 'undefined' || window.location.pathname !== '/checkout') {
      // TODO: make this dynamic in future
      return <>
        <a>Estimated shipping cost: &pound;3.45</a>
      </>
    } else {
      return <>
         <a>Shipping cost:</a>
       </>
    }
  }

  render () {
    const totals = calculateCartSummary(this.props.cart)

    return (
      <>
      <section className='c-cart-summary'>
        <dl>
          <dt><a>Total Price:</a></dt>
          <dd><a>&pound;{fixedPrice(totals.subTotal)}</a></dd>
        </dl>
        <dl>
          <dt>{ this.shippingText() }</dt>
        </dl>
        <dl>
          <dt><a className='u-bold'>TOTAL:</a></dt>
          <dd><a className='u-bold'>&pound;{fixedPrice(totals.total)}</a></dd>
        </dl>
        <dl>
          <a>* Including VAT</a>
        </dl>
      </section>
      <section className='c-cart-summary__buttons'>
        <Button
          aria-label='Continue Shopping'
          label='Continue shopping'
          size='lrg'
          status='grey'
          className='c-cart-summary__buttons--continue'
          type='button'
        />
        <Link href='/checkout' as=''>
          <Button
            aria-label='Place Order'
            label='Continue to payment'
            size='lrg'
            className='c-cart-summary__buttons--proceed'
            status='positive'
            type='button'
          />
        </Link>
      </section>
      </>
    )
  }
}

export default CartSummary
