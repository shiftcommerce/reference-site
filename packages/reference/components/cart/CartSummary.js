import { Component } from 'react'

export default class CartSummary extends Component {
  calculateSummary () {
    const lineItems = this.props.cart.lineItems
    let [ total, discount, tax, subTotal ] = [ 0, 0, 0, 0 ]

    lineItems.map((li, index) => {
      subTotal += (li.quantity * li.price)
      discount += (li.quantity * (li.discount || 0))
    })
    total = subTotal - discount + tax
    return { subTotal, total, tax, discount }
  }

  render () {
    const { subTotal, total, tax, discount } = this.calculateSummary()

    return (
      <section className='c-cart-summary'>
        <dl>
          <dt> Sub total: </dt>
          <dd> &pound;{ subTotal } </dd>
        </dl>
        <dl>
          <dt> Tax applied: </dt>
          <dd className='u-text-color--red'> +&pound;{ tax } </dd>
        </dl>
        <dl>
          <dt> Amount you saved on this purchase: </dt>
          <dd className='u-text-color--green'> -&pound;{ discount } </dd>
        </dl>
        <dl>
          <dt> Total amount to be paid: </dt>
          <dd> <b>&pound;{ total }</b> </dd>
        </dl>
      </section>
    )
  }
}
