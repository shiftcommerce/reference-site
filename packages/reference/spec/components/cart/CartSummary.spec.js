import CartSummary from '../../../components/cart/CartSummary'

test('renders correct values when no line items are available', () => {
  // arrange
  const cart = {
    lineItems: []
  }

  // act
  const wrapper = mount(
    <CartSummary cart={cart} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Sub total:  £0')
  expect(wrapper).toIncludeText('Tax applied:  +£0')
  expect(wrapper).toIncludeText('Amount you saved on this purchase:  -£0')
  expect(wrapper).toIncludeText('Total amount to be paid:  £0')
})

test('renders correct values when line items are available', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: 10,
        discount: 2,
        quantity: 2,
        sku: '123',
        imageUrl: '',
        size: 'size - 8'
      }
    ]
  }

  const updateQuantity = () => {}

  // act
  const wrapper = mount(
    <CartSummary cart={cart} updateQuantity={updateQuantity} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Sub total:  £20')
  expect(wrapper).toIncludeText('Tax applied:  +£0')
  expect(wrapper).toIncludeText('Amount you saved on this purchase:  -£4')
  expect(wrapper).toIncludeText('Total amount to be paid:  £16')
})
