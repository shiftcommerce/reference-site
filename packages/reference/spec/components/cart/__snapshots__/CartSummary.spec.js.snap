// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct values when line items are available 1`] = `
<CartSummary
  cart={
    Object {
      "lineItems": Array [
        Object {
          "discount": 2,
          "imageUrl": "",
          "price": 10,
          "quantity": 2,
          "size": "size - 8",
          "sku": "123",
          "title": "test",
        },
      ],
    }
  }
  updateQuantity={[Function]}
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
         Sub total: 
      </dt>
      <dd>
         £
        20
         
      </dd>
    </dl>
    <dl>
      <dt>
         Tax applied: 
      </dt>
      <dd
        className="u-text-color--red"
      >
         +£
        0
         
      </dd>
    </dl>
    <dl>
      <dt>
         Amount you saved on this purchase: 
      </dt>
      <dd
        className="u-text-color--green"
      >
         -£
        4
         
      </dd>
    </dl>
    <dl>
      <dt>
         Total amount to be paid: 
      </dt>
      <dd>
         
        <b>
          £
          16
        </b>
         
      </dd>
    </dl>
  </section>
</CartSummary>
`;

exports[`renders correct values when no line items are available 1`] = `
<CartSummary
  cart={
    Object {
      "lineItems": Array [],
    }
  }
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
         Sub total: 
      </dt>
      <dd>
         £
        0
         
      </dd>
    </dl>
    <dl>
      <dt>
         Tax applied: 
      </dt>
      <dd
        className="u-text-color--red"
      >
         +£
        0
         
      </dd>
    </dl>
    <dl>
      <dt>
         Amount you saved on this purchase: 
      </dt>
      <dd
        className="u-text-color--green"
      >
         -£
        0
         
      </dd>
    </dl>
    <dl>
      <dt>
         Total amount to be paid: 
      </dt>
      <dd>
         
        <b>
          £
          0
        </b>
         
      </dd>
    </dl>
  </section>
</CartSummary>
`;
